plugins/jenkins/app/models/samson/jenkins.rb

```diff
@@ -62,21 +66,31 @@ def jenkins_job_config
     # manually. To do this, we check if this job is mentioned in
     # description of jenkins job. if not, we add it. We use JENKINS_DESC_HEADER/FOOTER markers
     # to keep track of description added through samson.
-    def check_and_build_job_config
+    def check_job_config
       conf = jenkins_job_config
-      post_config = false
+      changes = {}
       expected_build_parameters = JENKINS_BUILD_PARAMETERS.keys
       present_build_params = build_params(conf).map(&:to_sym)
       missing_build_params = expected_build_parameters.to_set - present_build_params.to_set
       if missing_build_params.any?
-        add_build_parameters(conf, missing_build_params)
-        post_config = true
+        changes["build_params"] = missing_build_params
+      end
+      unless description_exists?(conf)
+        changes['job_description'] = true
+      end
+      changes
+    end
+
+    def build_job_config(changes)
+      Rails.cache.delete(job_name + JENKINS_JOB_CACHE_SUFFIX)
```

Comment:
these 2 could usages could get out of sync ... prefer a cache_key method ... don't need that constant then either